test/integration_test/migration_features_test.go

```diff
@@ -679,3 +681,109 @@ func testFailbackWithStashStrategy(t *testing.T) {
 	testResult = testResultPass
 	log.InfoD("Test status at end of %s tests: %s", t.Name(), testResult)
 }
+
+func validateCRWithTransformationValue(appName string, namespace string, transfomration *storkapi.ResourceTransformation, kubeConfig string) error {
+	operatorCRMap := getSupportedOperatorCRMapping()
+	if _, ok := operatorCRMap[appName]; !ok {
+		return fmt.Errorf("app %s is not currently supported in test framework", appName)
+	}
+	resources := operatorCRMap[appName]
+
+	config, err := clientcmd.BuildConfigFromFlags("", kubeConfig)
+	if err != nil {
+		return fmt.Errorf("error building config from kubeconfig failed: %v", err)
+	}
+	resourceCollector := resourcecollector.ResourceCollector{
+		Driver: nil,
+	}
+	err = resourceCollector.Init(config)
+	if err != nil {
+		return fmt.Errorf("error initing resourcecollector while validating CR: %v", err)
+	}
+
+	for _, resource := range resources {
+		objects, _, err := resourceCollector.GetResourcesForType(
+			resource,
+			nil,
+			[]string{namespace},
+			nil,
+			nil,
+			nil,
+			false,
+			resourcecollector.Options{},
+		)
+		if err != nil {
+			return fmt.Errorf("error fetching objects while validating CR: %v", err)
+		}
+		log.InfoD("number of resources for app %s in namespace %s is %d", appName, namespace, len(objects.Items))
+		for _, object := range objects.Items {
+			for _, specObject := range transfomration.Spec.Objects {
+				for _, resourcePath := range specObject.Paths {
+					// TODO: Handle the case for delete operation when required in test validation
```

Comment:
Deletepath will require more code to check if the field itself has been deleted, with the field removal it can remove the it's parent key too if that's the only key in a nested object. It needs more code to handle that validation.